File: apis/backend/src/main/java/com/tms/api/objects/models/sprint/SprintTaskModel.java
package com.tms.api.objects.models.sprint;

import com.tms.api.objects.models.user.sprint.TaskUserDetailModel;
import com.tms.api.objects.types.task.TaskType;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class SprintTaskModel {
    private String taskId;
    private String description;
    private TaskType taskType;
    private TaskUserDetailModel assignee;
    private TaskUserDetailModel owner;
}

File: ticket-managment-system-web/src/app/core/models/sprint/sprint-task.model.ts
import { TaskType } from "../../enums";
import { TaskUserDetailModel } from "../user";

export class SprintTaskModel {
    public taskId!: String;
    public description!: String;
    public taskType!: TaskType;
    public assignee: TaskUserDetailModel | null = null;
    public owner!: TaskUserDetailModel;


    constructor(sprintTask: Partial<SprintTaskModel> = {}) {
        Object.assign(this, sprintTask);
    }
}